src/test/scala/org/ergoplatform/network/ErgoNodeViewSynchronizerSpecification.scala

```diff
@@ -236,37 +235,87 @@ class ErgoNodeViewSynchronizerSpecification extends HistoryTestHelpers with Matc
+        deliveryTracker.setRequested(Seq(header.id), Header.modifierTypeId, Some(peer))(_ => Cancellable.alreadyCancelled)
+        val modData = ModifiersData(Header.modifierTypeId, Map(header.id -> header.bytes))
+        val modSpec = new ModifiersSpec(100)
+        synchronizerMockRef ! Message(modSpec, Left(modSpec.toBytes(modData)), Some(peer))
+      }
+
+      deliveryTracker.reset()
+
+      // we generate and apply existing base chain
+      val hhistory = ErgoHistory.readOrGenerate(settings, timeProvider)
+      val baseChain = genHeaderChain(_.size > 4, None, hhistory.difficultyCalculator, None, false)
+      baseChain.headers.foreach(sendHeader)
+      val bestHeader =
+        eventually {
+          val bestHeaderOpt = hhistory.bestHeaderOpt
+          assert(bestHeaderOpt.nonEmpty)
+          bestHeaderOpt
+        }
+
+      // then a continuation chain that will be part of the syncV2 message
+      val continuationChain = genHeaderChain(_.size > 4, bestHeader, hhistory.difficultyCalculator, None, false)
+
+      // sync message carries best header of our base change + continuation chain whose Head header is supposed to be applied
+      val sync = ErgoSyncInfoV2((bestHeader.get +: continuationChain.headers).reverse)
+      val msgBytes = ErgoSyncInfoMessageSpec.toBytes(sync)
+
+      // send this sync msg to synchronizer which should apply the header following the common header from base chain
+      synchronizerMockRef ! Message(ErgoSyncInfoMessageSpec, Left(msgBytes), Some(peer))
+      eventually {
+        val expectedHeaderId = continuationChain.head.id
+        hhistory.bestHeaderOpt.get.id shouldBe expectedHeaderId
+      }
+    }
+  }
+
   property("NodeViewSynchronizer: receiving out-of-order header should request it again") {
     withFixture2 { ctx =>
       import ctx._
 
       implicit val patienceConfig: PatienceConfig = PatienceConfig(5.seconds, 100.millis)
 
-      def sendHeader(synchronizerRef: ActorRef,  header: Header): Unit = {
+      def sendHeader(header: Header): Unit = {
         deliveryTracker.setRequested(Seq(header.id), Header.modifierTypeId, Some(peer))(_ => Cancellable.alreadyCancelled)
         val modData = ModifiersData(Header.modifierTypeId, Map(header.id -> header.bytes))
         val modSpec = new ModifiersSpec(100)
-        synchronizerRef ! Message(modSpec, Left(modSpec.toBytes(modData)), Some(peer))
+        synchronizerMockRef ! Message(modSpec, Left(modSpec.toBytes(modData)), Some(peer))
       }
 
       deliveryTracker.reset()
 
       // we generate fork of two headers, starting from the parent of the best header
       // so the depth of the rollback is 1, and the fork bypasses the best chain by 1 header
       val hhistory = ErgoHistory.readOrGenerate(settings, timeProvider)
-      val parent = hhistory.lastHeaders(2).head
+      val baseChain = genHeaderChain(_.size > 2, None, hhistory.difficultyCalculator, None, false)
+      baseChain.headers.foreach(sendHeader)
```

Comment:
@kushti I made this change in out-of-order test as it fails on CI with empty `.ergo_test/history` folder ... it only works if some preceding tests generates some headers apriory.